ovs/lib/helpers/vpool/shared.py

```diff
@@ -38,3 +43,79 @@ def retrieve_alba_arakoon_config(alba_backend_guid, ovs_client):
         return arakoon_config
+
+    @staticmethod
+    def retrieve_local_alba_arakoon_config(alba_backend_guid, as_ini=False):
+        # type: (str, Optional[bool]) -> dict
+        """
+        Retrieves the local ALBA Arakoon configuration.
+        WARNING: YOU DO NOT BELONG HERE, PLEASE MOVE TO YOUR OWN PLUGIN
+        :param alba_backend_guid: Guid of the ALBA Backend
+        :type alba_backend_guid: str
+        :param as_ini: Save the config as an INI file for alba
+        :type as_ini: bool
+        :return: Arakoon configuration information
+        :rtype: dict
+        """
+        cfg = Configuration.get(REMOTE_CONFIG_BACKEND_CONFIG.format(alba_backend_guid), default=None)
+        if as_ini:
+            cfg = ArakoonClusterConfig.convert_config_to(cfg, return_type='INI')
+        return cfg
+
+    @classmethod
+    def sync_alba_arakoon_config(cls, alba_backend_guid, ovs_client):
+        # type: (str, OVSClient) -> None
+        """
+        Compares the remote and local config. Updates the local config if needed. Guarantees the latest greatest config
+        WARNING: YOU DO NOT BELONG HERE, PLEASE MOVE TO YOUR OWN PLUGIN
+        :param alba_backend_guid: Guid of the ALBA Backend
+        :type alba_backend_guid: str
+        :param ovs_client: OVS client object
+        :type ovs_client: OVSClient
+        :return: Arakoon configuration information
+        :rtype: dict
+        """
+        #todo: Fix leak of backends not bein removed
+
+        remote_config = cls._retrieve_remote_alba_arakoon_config(alba_backend_guid, ovs_client)
+        current_config = Configuration.get(REMOTE_CONFIG_BACKEND_CONFIG.format(alba_backend_guid), default=None)
+        if current_config != remote_config:
+            transaction_id = Configuration.begin_transaction()
+            Configuration.set(key=REMOTE_CONFIG_BACKEND_CONFIG.format(alba_backend_guid), value=remote_config, transaction=transaction_id)
+
+            ini_config = ArakoonClusterConfig.convert_config_to(config=remote_config, return_type='INI')
+            Configuration.set(key=REMOTE_CONFIG_BACKEND_INI.format(alba_backend_guid), value=ini_config, raw=True, transaction=transaction_id)
+            Configuration.apply_transaction(transaction_id)
+
+
+    @classmethod
+    def calculate_abm_configs_in_use(cls):
+        """
+        Iterate over all vPools in the cluster and check which local or remote backend configs are still in use.
+        :return:
+        """
+        present_remote_configs = dict([(key, REMOTE_CONFIG_BACKEND_CONFIG.format(key)) for key in list(Configuration.list(REMOTE_CONFIG_BACKEND_BASE))])
+        in_use = set()
+        cache_types = [StorageDriverConfiguration.CACHE_FRAGMENT, StorageDriverConfiguration.CACHE_BLOCK]
+
+        for vpool in VPoolList.get_vpools():
+            print vpool.name
+            proxy_config_template = '/ovs/vpools/{0}/proxies/{{0}}/config/main'.format(vpool.guid)
```

Comment:
Can we keep this constant in a constants file?

Reply:
https://github.com/openvstorage/framework-extensions/pull/194